text/0000-imply-option.md

```diff
@@ -0,0 +1,139 @@
+- Feature Name: imply-option
+- Start Date: 2017-10-18
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This is an RFC to reduce common boiler plate code when making use of the `Option` type. Similar in intention and motivation to the `Try` trait for `Result`.
+
+# Motivation
+[motivation]: #motivation
+
+This addition will increase the legibility of code segments and assist in defining the thought processes and motivations of programmers through code. The use cases of this addition are solutions which are expressable in the following predicate form:
+```
+    P(x) : Predicate on `x`.
+    F(y) : Function on `y`
+    P(x) -> F(y)
```

Comment:
Is this a new proposed syntax? It sounds like you're introducing a mathematical syntax which you're going to use later to talk about something else.

Reply:
This is not a proposed syntax. It is predicate logic which is the syntax of Boolean algebra. I am using it as a tool to formally define my solution rather than only waving my hands around and providing examples.